homeassistant/components/yale_smart_alarm/coordinator.py

```diff
@@ -0,0 +1,118 @@
+"""DataUpdateCoordinator for the Yale integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+
+from yalesmartalarmclient.client import AuthenticationError, YaleSmartAlarmClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    STATE_LOCKED,
+    STATE_UNAVAILABLE,
+    STATE_UNLOCKED,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN, LOGGER
+
+
+class YaleDataUpdateCoordinator(DataUpdateCoordinator):
+    """A Yale Data Update Coordinator."""
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize the Verisure hub."""
+        self.entry = entry
+        self._hass = hass
+        self.yale = None
+
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=DEFAULT_SCAN_INTERVAL),
+        )
+
+    async def _async_update_data(self) -> dict:
+        """Fetch data from Yale."""
+
+        if self.yale is None:
+            self.yale = await self._hass.async_add_executor_job(
```

Comment:
instead of wrapping each function call in a job, create a synchronous function that wraps all the synchronous function calls and call that function in an executor job from your asynchronous update function

Reply:
Changed as suggested